### PR TITLE
adding unit test for device plugin

### PR DESCRIPTION
**Pull request by crystalzhaizhai**

### CODE REVIEW

File: pkg/gpu/nvidia/metrics/metrics.go [hunk 1]
```diff
@@ -26,6 +26,29 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+type metricsCollector interface {
+	collectGPUDevice(deviceName string) (*nvml.Device, error)
+	collectStatus(*nvml.Device) (status *nvml.DeviceStatus, err error)
+	collectDutyCycle(string, time.Duration) (uint, error)
+}
+
+var g metricsCollector
```
Comment (grac3gao) [inline pkg/gpu/nvidia/metrics/metrics.go:35]: nit: maybe a longer name for this package level global variable?

File: pkg/gpu/nvidia/metrics/metrics_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+package metrics
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	//"github.com/golang/glog"
```
[the remark this answers is not in the document]
Comment (crystalzhaizhai) [reply pkg/gpu/nvidia/metrics/metrics_test.go]: done

File: pkg/gpu/nvidia/metrics/metrics.go [hunk 3]
```diff
@@ -26,6 +26,29 @@ import (
...
+
+type mCollector struct{}
+
+func (t *mCollector) collectGPUDevice(deviceName string) (*nvml.Device, error) {
+	return DeviceFromName(deviceName)
+}
+
+func (t *mCollector) collectStatus(d *nvml.Device) (status *nvml.DeviceStatus, err error) {
+	status, err = d.Status()
```
[the remark this answers is not in the document]
Comment (crystalzhaizhai) [reply pkg/gpu/nvidia/metrics/metrics.go:46]: I think that works. I don't remember any other purposes.

File: pkg/gpu/nvidia/metrics/metrics.go (hunk 1 above)
Comment (crystalzhaizhai) [reply pkg/gpu/nvidia/metrics/metrics.go]: changed to gmc...not good with names...Do you have better suggestions?

Comment (grac3gao) [reply pkg/gpu/nvidia/metrics/metrics.go]: gmc looks good to me

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (grac3gao) [inline .vscode/launch.json:1]: Exclude this vscode related file from your patch?

**Review by grac3gao — APPROVED** [review]

**Pull request merged**